src/records.jl

```diff
@@ -157,7 +157,7 @@ function DefaultRecord(name::AbstractString, level::AbstractString, levelnum::In
     trace = Attribute{StackTrace}(get_trace)
 
     DefaultRecord(
-        Attribute{ZonedDateTime}(() -> Dates.now(tz"UTC")),
```

Comment:
Why not default to `now(UTC)` (which gives a `DateTime`)? Is the current default local time so we want to preserve that?

Reply:
Yeah, I did this to preserve the default output behaviour. The previous logic was to convert the `ZonedDateTime(dt, tz"UTC")` to `localzone()` if no `output_tz` is provided. If we did `now(UTC)`, then we'd always need TimeZones.jl loaded to convert that back to a localzone in the default case.

Current behaviour:
```julia
julia> info(logger, "Something you might want to know")
[2021-11-10 09:44:21 | info | root]: Something you might want to know
```

Current PR:
```julia
julia> info(logger, "Something you might want to know")
[2021-11-10 09:46:14 | info | root]: Something you might want to know
```

UTC Change:
```julia
julia> info(logger, "Something you might want to know")
[2021-11-10 17:48:17 | info | root]: Something you might want to know
```

Reply:
I guess my one concern is that this could break specialized formatters which process the raw records. That being said, this will likely break more explicitly since we're not storing a `ZonedDateTime` and they can also modify the formatting code to assume localzone?

Reply:
I'm frustrated that the Dates stdlib has a notion of local and UTC but can't convert between them. But given that it can't, I think this is probably the best choice.

Reply:
Yeah, I was considering updating the Dates stdlib with that basic conversion, but that seemed like a bit of a rabbit hole. The fact that everything else seems to default to `localzone` makes me thing that's the best default here.